src/components/navigation/GuideKeyHandler.ts

```diff
@@ -17,56 +17,45 @@ export interface ExtendedHTMLElement extends HTMLElement {
  * @param {React.KeyboardEvent} event - The keyboard event.
  * @param {React.RefObject<HTMLElement>} chartRef - Reference to the chart element.
  */
-export function guideKeyHandler(
-	event: React.KeyboardEvent,
-	chartRef: React.RefObject<HTMLElement>,
-): void {
+export function guideKeyHandler({
+	event,
+	chartRef,
+	setVisibleShortcutGuide,
+	shortcutGuideRef,
+}: {
+	event: React.KeyboardEvent;
+	chartRef: React.RefObject<HTMLElement>;
+	setVisibleShortcutGuide: Function;
+	shortcutGuideRef: React.RefObject<HTMLElement>;
+}): void {
 	const { key } = event;
-	const activeElement = document.activeElement as HTMLElement;
 
-	const shouldHandleKey =
-		activeElement?.classList.contains("a11y_modal_content") ||
-		activeElement?.classList.contains("a11y_row") ||
-		activeElement?.id === "guide_close";
+	const shouldHandleKey = shortcutGuideRef.current?.contains(document.activeElement);
```

Comment:
I believe this should be refactored. A piece of state in the `AutoVizua11y` should hold a boolean that represents the Dialog being open or not. Then it should be used as a prop or an argument.